src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs

```diff
@@ -70,7 +70,7 @@ internal static T[] ToArray<T>(IEnumerable<T> source, out int length)
                                 int newLength = count << 1;
                                 if ((uint)newLength > MaxArrayLength)
                                 {
-                                    newLength = MaxArrayLength <= count ? count + 1 : MaxArrayLength;
+                                    newLength = count >= MaxArrayLength ? count + 1 : MaxArrayLength;
```

Comment:
I thought this change was only about equality operators. I am not sure I see the value of reordering inequality operators.

Reply:
SA1131 was flagged on this part of the code, so it was changed.

Reply:
OK, fair enough. I guess my suggestion is as above then, to bite off a smaller part of the tree, and get it clean for SA1131. You could force-push that smaller change to this PR maybe.